### PR TITLE
AE: improve tempo filter

### PR DESCRIPTION
**Pull request by FernetMenta**
Improves accuracy of tempo filter

### CODE REVIEW

**Comment by FernetMenta** [comment]
jenkins build this please

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEFilter.cpp [hunk 1]
```diff
@@ -277,6 +280,8 @@ int CActiveAEFilter::ProcessFilter(uint8_t **dst_buffer, int dst_samples, uint8_
       return -1;
     }
 
+    m_SamplesOut = m_pOutFrame->pts;
```
Comment (popcornmix) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEFilter.cpp]: I think this is responsible for the hang on Pi when pts=AV_NOPTS_VALUE.

Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEFilter.cpp]: did you log pts? pts of outFrame is independent of input in this case. it should never be equal to AV_NOPTS_VALUE. further why should this behave different on Pi?

Comment (popcornmix) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEFilter.cpp]: Yes, I added this logging:
http://paste.ubuntu.com/24450547/
and got this log:
http://paste.ubuntu.com/24450549/
Certainly looks like pts=AV_NOPTS_VALUE

Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEFilter.cpp]: oh, this should be outFrame, not m_pOutFrame. if format conversion is required outFrame is set to m_pConvertFrame. that is the reason why the issue showed on Pi and not on Ubuntu

**Comment by popcornmix** [comment]
Updated version doesn't hang any more on Pi.

**Comment by FernetMenta** [comment]
jenkins build this please

**Pull request merged**